src/cigar.c

```diff
@@ -13,7 +13,14 @@ void edits_to_cigar(const char *edits, char **cigar)
 
 static inline int count_edits(const char *cigar)
 {
-    return 42; // Work out how long the string will be
+    int n = strlen(cigar);
+    int len = 0;
+    for (int i = 0 ; i < n ; i++ ) {
+        if (i % 2 == 0 ) {
+            len += atoi(&cigar[i]);
+        }
```

Comment:
I think you are attempting to get every second character as an integer. It looks that way, at least. This will work, if all the integers are single-digit, but they might not be. This is a valid cigar: "12M2I". Your code will call `atoi` on "12M" (`cigar + 0 == &cigar[0]`) and get 12, which is fine, but then you call on "M2I" which is zero, and then finally on "I" which is also zero. The result is 12, but it should be 14.